src/data-model-providers/codegen/CodegenDataModelProvider.cpp

```diff
@@ -343,6 +343,12 @@ std::optional<DataModel::ActionReturnStatus> CodegenDataModelProvider::Invoke(co
         {
             return std::nullopt;
         }
+        else
+        {
+            // If we reach here, it means this particular CommandHandlerInterface recognized the cluster but the command is invalid or cannot be processed.
+            handler->AddStatus(request.path, Protocols::InteractionModel::Status::InvalidCommand);
```

Comment:
We previously allowed fallback to ember functions ... so this is not 100% backwards compatible (then again I am unsure if we should be or not).

I would lean more towards passing in a `what to return on no handler found` to the dispatch handler instead.

Reply:
But when it is fall-backed to the DispatchSingleClusterCommand, it just calls the following function to add status, what is the difference if we just add the status here?

```
    if (errorStatus != Protocols::InteractionModel::Status::Success)
    {
        apCommandObj->AddStatus(aCommandPath, errorStatus);
    }
```